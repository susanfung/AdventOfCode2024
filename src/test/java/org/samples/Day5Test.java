package org.samples;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

public class Day5Test {
    @Test
    public void testDay5Part1_parseRules() throws IOException {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5TestData.txt";

        BufferedReader reader = new BufferedReader(new FileReader(filePath));

        day5.parseRules(reader);

        Approvals.verify(day5.rules);
    }

    @Test
    public void testDay5Part1_isValidOrder() throws IOException {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5TestData.txt";

        BufferedReader reader = new BufferedReader(new FileReader(filePath));

        day5.parseRules(reader);

        String update = "75,47,61,53,29";

        int[] pages = Arrays.stream(update.split(","))
                            .mapToInt(Integer::parseInt)
                            .toArray();

        Approvals.verify(day5.isValidOrder(pages));
    }

    @Test
    public void testDay5Part1_solvePart1() {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5TestData.txt";

        int result = day5.solvePart1(filePath);

        Approvals.verify(result);
    }

    @Test
    public void testDay5Part1_answer() {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5Data.txt";

        int result = day5.solvePart1(filePath);

        Approvals.verify(result);
    }

    @Test
    public void testDay5Part2_solvePart2() {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5TestData.txt";

        int result = day5.solvePart2(filePath);

        Approvals.verify(result);
    }

    @Test
    public void testDay5Part2_answer() {
        Day5 day5 = new Day5();
        String filePath = "src/test/java/org/samples/Day5Data.txt";

        int result = day5.solvePart2(filePath);

        Approvals.verify(result);
    }
}
