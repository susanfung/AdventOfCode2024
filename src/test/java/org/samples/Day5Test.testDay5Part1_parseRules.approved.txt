29 : [13]
47 : [53, 13, 61, 29]
53 : [29, 13]
61 : [53, 13, 29]
75 : [53, 29, 61, 13, 47]
97 : [53, 75, 13, 61, 29, 47]
